src/MyersDiff.php

```diff
@@ -73,11 +73,13 @@ private function extractSnakes(array $v_save, $x, $y)
     /**
      * Convert a list of "snakes" into a set of insert/keep/delete instructions.
      *
-     * @param integer[][] $snakes Common subsequences
-     * @param string[]    $a      First sequence
-     * @param string[]    $b      Second sequence
+     * @template T
      *
-     * @return array[] - pairs of token and edit (-1 for delete, 0 for keep, +1 for insert)
+     * @param list<array{int, int}> $snakes Common subsequences
+     * @param list<T>               $a      First sequence
+     * @param list<T>               $b      Second sequence
+     *
+     * @return list<array{T, -1,0,1}> - pairs of token and edit (-1 for delete, 0 for keep, +1 for insert)
```

Comment:
For the record:
I am used to Drupal doc style which I find more readable:
```suggestion
     * @param list<array{int, int}> $snakes
     *   Common subsequences.
     * @param list<T> $a
     *   First sequence.
     * @param list<T> $b
     *   Second sequence.
     *
     * @return list<array{T, -1|0|1}>
     *   Pairs of token and edit (-1 for delete, 0 for keep, +1 for insert)
```